### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: modules/remotebackend/pipeconnector.cc [hunk 1]
```diff
@@ -142,7 +142,7 @@ int PipeConnector::send_message(const Json& input)
   unsigned int sent = 0;
   int bytes;
 
-  // writen routine - socket may not accept al data in one go
+  // written routine - socket may not accept al data in one go
```
Comment (pieterlexis) [inline modules/remotebackend/pipeconnector.cc:145]: `writen` is valid.. it is a `write()` that writes `n` bytes

**Review by rgacogne — CHANGES_REQUESTED** [review]
I don't know what happened but some symbolic links have been turned into a copy of the target, duplicating code.

**Comment by jsoref** [comment]
Sigh, one of my tools is stupid. I'll find the commits that need to be undone.

**Review by rgacogne — APPROVED** [review]
LGTM, nice work!

File: pdns/iputils.hh [hunk 2]
```diff
@@ -1276,7 +1276,7 @@ private:
   size_type d_size;
 };
 
-/** This class represents a group of supplemental Netmask classes. An IP address matchs
+/** This class represents a group of supplemental Netmask classes. An IP address matches
     if it is matched by zero or more of the Netmask classes within.
```
Comment (Habbie) [inline pdns/iputils.hh:1280]:
```suggestion
    if it is matched by one or more of the Netmask classes within.
```

File: pdns/lua-record.cc [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
    ponder netmask tree from file for huge number of netmasks
 
-   unify ifupurl/ifupport
+   unify ifupurl/ifportup
```
Comment (Habbie) [inline pdns/lua-record.cc:28]:
```suggestion
   unify ifurlup/ifportup
```

File: pdns/test-dnsdist_cc.cc [hunk 4]
```diff
@@ -42,7 +42,7 @@ BOOST_AUTO_TEST_SUITE(test_dnsdist_cc)
 static const uint16_t ECSSourcePrefixV4 = 24;
 static const uint16_t ECSSourcePrefixV6 = 56;
 
-static void validateQuery(const PacketBuffer& packet, bool hasEdns=true, bool hasXPF=false, uint16_t additionals=0, uint16_t answers=0, uint16_t authorities=0)
+static void validateQuery(const PacketBuffer& packet, bool hasEdns=true, bool hasXPF=false, uint16_t additional=0, uint16_t answers=0, uint16_t authorities=0)
```
Comment (Habbie) [inline pdns/test-dnsdist_cc.cc:45]:
```suggestion
static void validateQuery(const PacketBuffer& packet, bool hasEdns=true, bool hasXPF=false, uint16_t additionals=0, uint16_t answers=0, uint16_t authorities=0)
```

File: pdns/test-dnsdist_cc.cc [hunk 5]
```diff
@@ -51,7 +51,7 @@ static void validateQuery(const PacketBuffer& packet, bool hasEdns=true, bool ha
   BOOST_CHECK_EQUAL(mdp.d_header.qdcount, 1U);
   BOOST_CHECK_EQUAL(mdp.d_header.ancount, answers);
   BOOST_CHECK_EQUAL(mdp.d_header.nscount, authorities);
-  uint16_t expectedARCount = additionals + (hasEdns ? 1U : 0U) + (hasXPF ? 1U : 0U);
+  uint16_t expectedARCount = additional + (hasEdns ? 1U : 0U) + (hasXPF ? 1U : 0U);
```
Comment (Habbie) [inline pdns/test-dnsdist_cc.cc:54]:
```suggestion
  uint16_t expectedARCount = additionals + (hasEdns ? 1U : 0U) + (hasXPF ? 1U : 0U);
```

File: regression-tests/tests/too-big-for-udp-query-always-truncate-additional/description [hunk 6]
```diff
@@ -1,6 +1,6 @@
 UDP dns packets can only be 512 bytes long - when they are longer, they need
 to get truncated, and have the 'TC' bit set, to inform the client that they
 need to requery over TCP. This query however does not need truncation, since
-the information that causes things to go over limit are 'courtesy' additionals.
+the information that causes things to go over limit are 'courtesy' additional.
```
Comment (Habbie) [inline regression-tests/tests/too-big-for-udp-query-always-truncate-additional/description:4]:
```suggestion
the information that causes things to go over limit are 'courtesy' additionals.
```

**Review by Habbie — CHANGES_REQUESTED** [review]
Thanks for this! I posted 5 commit suggestions. 3 of them are reversing a mistaken change from 'additionals' to 'additional'. The other 2 are errors I noticed that you did not cause but are so close to your work that we might as well do them now - also if I PRd those now they would conflict with your work.

If you take all 5 commits, I'll approve this.

**Pull request merged**